src/pkg/services/m365/api/graph/middleware.go

```diff
@@ -164,7 +164,15 @@ func (mw RetryMiddleware) Intercept(
 		mw.isRetriableRespCode(ctx, resp)
 
 	if !retriable {
-		return resp, stackReq(ctx, req, resp, err).OrNil()
+		// Returning a response and an error causes output from either some part of
+		// the middleware/graph/golang or the mocking library used for testing.
+		// Return one or the other to avoid this.
+		err := stackReq(ctx, req, resp, err).OrNil()
+		if err != nil {
```

Comment:
Unrelated to this PR, we can simplify this function a bit. For successful requests(`err == nil`), we can return early after line 160. Currently we are using `!retriable` as a way to also catch successful requests which is not super ideal.